contracts/contracts/strategies/MorphoAaveStrategy.sol

```diff
@@ -0,0 +1,239 @@
+        address[] calldata _pTokens
+    ) external onlyGovernor initializer {
+        super._initialize(
+            MORPHO,
+            _vaultAddress,
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+    }
+
+    /**
+     * @dev Approve the spending of all assets by main Morpho contract,
+     *      if for some reason is it necessary.
+     */
+    function safeApproveAllTokens()
+        external
+        override
+        onlyGovernor
+        nonReentrant
+    {
+        uint256 assetCount = assetsMapped.length;
+        for (uint256 i = 0; i < assetCount; i++) {
+            address asset = assetsMapped[i];
+
+            // Safe approval
+            IERC20(asset).safeApprove(MORPHO, 0);
+            IERC20(asset).safeApprove(MORPHO, type(uint256).max);
+        }
+    }
+
+    /**
+     * @dev Internal method to respond to the addition of new asset
+     *      We need to approve and allow Morpho to move them
+     * @param _asset Address of the asset to approve
+     * @param _pToken The pToken for the approval
+     */
+    // solhint-disable-next-line no-unused-vars
+    function _abstractSetPToken(address _asset, address _pToken)
+        internal
+        override
+    {
+        IERC20(_asset).safeApprove(MORPHO, 0);
+        IERC20(_asset).safeApprove(MORPHO, type(uint256).max);
+    }
+
+    /**
+     * @dev Collect accumulated rewards and send them to Harvester.
+     */
+    function collectRewardTokens()
+        external
+        override
+        onlyHarvester
+        nonReentrant
+    {
+        // Morpho Aave-v2 doesn't distribute reward tokens
+        // solhint-disable-next-line max-line-length
+        // Ref: https://developers.morpho.xyz/interact-with-morpho/get-started/interact-with-morpho/claim-rewards#morpho-aave-v2
+    }
+
+    /**
+     * @dev Get the amount of rewards pending to be collected from the protocol
+     */
+    function getPendingRewards() external view returns (uint256 balance) {
+        // Morpho Aave-v2 doesn't distribute reward tokens
+        // solhint-disable-next-line max-line-length
+        // Ref: https://developers.morpho.xyz/interact-with-morpho/get-started/interact-with-morpho/claim-rewards#morpho-aave-v2
+        return 0;
+    }
+
+    /**
+     * @dev Deposit asset into Morpho
+     * @param _asset Address of asset to deposit
+     * @param _amount Amount of asset to deposit
+     */
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        _deposit(_asset, _amount);
+    }
+
+    /**
+     * @dev Deposit asset into Morpho
+     * @param _asset Address of asset to deposit
+     * @param _amount Amount of asset to deposit
+     */
+    function _deposit(address _asset, uint256 _amount) internal {
+        require(_amount > 0, "Must deposit something");
+
+        IMorpho(MORPHO).supply(
+            address(_getPTokenFor(_asset)),
```

Comment:
`_deposit` calls `_getPTokenFor` twice. This does two storage reads. 

It could be more efferent to do one read into a temporary variable (as `_withdraw` does), then use that in the rest of the function.

Reply:
Agreed, have changed it